docs/source/data/partitioned_and_incremental_datasets.md

```diff
@@ -238,6 +239,24 @@ def create_partitions() -> Dict[str, Callable[[], Any]]:
 When using lazy saving, the dataset will be written _after_ the `after_node_run` [hook](../hooks/introduction).
 ```
 
+```{note}
+Lazy saving is a default behaviour, meaning that if `Callable` type provided the dataset will be written _after_ the `after_node_run` hook.
+```
+
+In some cases, it might be useful to disable such a behaviour, for example, when your object is already `Callable`, like a Tensorflow model, and you do not mean to save it lazily.
```

Comment:
```suggestion
In certain cases, it might be useful to disable lazy saving, such as when your object is already a `Callable` (e.g., a TensorFlow model) and you do not intend to save it lazily."
```